internal/oapispec/ffi2swagger.go

```diff
@@ -35,35 +38,101 @@ func NewFFISwaggerGen() FFISwaggerGen {
 	return &ffiSwaggerGen{}
 }
 
-func (og *ffiSwaggerGen) Generate(ctx context.Context, baseURL string, ffi *fftypes.FFI) (swagger *openapi3.T, err error) {
-
-	// routes := []*Route{}
-
-	// for _, method := range ffi.Methods {
-	// 	if routes, err = og.addMethod(ctx, routes, method); err != nil {
-	// 		return nil, err
-	// 	}
-	// }
-
-	// for _, event := range ffi.Events {
-	// 	if routes, err = og.addEvent(ctx, routes, event); err != nil {
-	// 		return nil, err
-	// 	}
-	// }
-
-	return nil, nil
+func (og *ffiSwaggerGen) Generate(ctx context.Context, baseURL string, api *fftypes.ContractAPI, ffi *fftypes.FFI) (swagger *openapi3.T, err error) {
+	hasLocation := api.Location.String() != "null"
```

Comment:
We have a constant for this, and probably should even go a step further and have an `IsNull()` call on the type... I think it'll cost in the long term to have this string-null checking too proliferated in the codebase.

Reply:
Agreed, this felt very wrong but was the only thing that worked (`api.Location == nil` and `len(api.Location) == 0` both did not work). I'll look at adding a helper.